example/pauli/stmts.py

```diff
@@ -0,0 +1,35 @@
+from numbers import Number
+
+import numpy as np
+
+from kirin import ir, types, lowering
+from kirin.decl import info, statement
+
+from .dialect import dialect_
+
+
+@statement
+class PauliOperator(ir.Statement):
+    traits = frozenset({ir.Pure(), lowering.FromPythonCall()})
+    pre_factor: Number = info.attribute(default=1)
+    result: ir.ResultValue = info.result(types.PyClass(np.matrix))
+
+
+@statement(dialect=dialect_)
+class X(PauliOperator):
+    pass
+
+
+@statement(dialect=dialect_)
+class Y(PauliOperator):
+    pass
+
+
+@statement(dialect=dialect_)
+class Z(PauliOperator):
+    pass
+
+
+@statement(dialect=dialect_)
+class Id(PauliOperator):
+    pass
```

Comment:
not sure if you want to also have it, but fyi. if you want to get rid of those linter error you can implement some dummy `lowering.wraps` to trick Python linter to believe your output is not an SSA but a `np.matrix`

Reply:
I know, but I didn't want to bother with the wraps for the example.